registry/storage/driver/s3-aws/s3.go

```diff
@@ -667,13 +667,13 @@ func (d *driver) List(ctx context.Context, opath string) ([]string, error) {
 			directories = append(directories, strings.Replace(commonPrefix[0:len(commonPrefix)-1], d.s3Path(""), prefix, 1))
 		}
 
-		if *resp.IsTruncated {
-			resp, err = d.S3.ListObjects(&s3.ListObjectsInput{
-				Bucket:    aws.String(d.Bucket),
-				Prefix:    aws.String(d.s3Path(path)),
-				Delimiter: aws.String("/"),
-				MaxKeys:   aws.Int64(listMax),
-				Marker:    resp.NextMarker,
+		if resp.IsTruncated != nil && *resp.IsTruncated {
+			resp, err = d.S3.ListObjectsV2(&s3.ListObjectsV2Input{
+				Bucket:            aws.String(d.Bucket),
+				Prefix:            aws.String(d.s3Path(path)),
+				Delimiter:         aws.String("/"),
+				MaxKeys:           aws.Int64(listMax),
+				ContinuationToken: resp.ContinuationToken,
```

Comment:
This must be set to `resp.NextContinuationToken`

As per [AWS docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html):

> NextContinuationToken
NextContinuationToken is sent when isTruncated is true, which means there are more keys in the bucket that can be listed. The next list requests to Amazon S3 can be continued with this NextContinuationToken. NextContinuationToken is obfuscated and is not a real key

Reply:
yes, you are right, I' ll fix it.

Reply:
commit again.
PTAL